File: src/main/java/com/server/computerscience/chatbot/dto/request/ChatGptRequestFileUploadDto.java
package com.server.computerscience.chatbot.dto.request;

import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Getter
@AllArgsConstructor
@NoArgsConstructor
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
@Builder
@ToString
public class ChatGptRequestFileUploadDto {
	private String customId;
	private String method;
	private String url;
	private ChatGptRestRequestDto body;

	public static ChatGptRequestFileUploadDto from(ChatGptRestRequestDto body,
		String customId,
		String method,
		String url) {
		return ChatGptRequestFileUploadDto.builder()
			.body(body)
			.customId(customId)
			.method(method)
			.url(url)
			.build();
	}
}

File: src/main/java/com/server/computerscience/chatbot/dto/request/ChatGptRequestFileUploadDto.kt
package com.server.computerscience.chatbot.dto.request

import com.fasterxml.jackson.databind.PropertyNamingStrategies
import com.fasterxml.jackson.databind.annotation.JsonNaming

@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy::class)
data class ChatGptRequestFileUploadDto(
    val customId: String? = null,
    val method: String? = null,
    val url: String? = null,
    val body: ChatGptRestRequestDto? = null
) {

    companion object {
        @JvmStatic
        fun from(
            body: ChatGptRestRequestDto?,
            customId: String?,
            method: String?,
            url: String?
        ): ChatGptRequestFileUploadDto {
            return ChatGptRequestFileUploadDto(
                customId = customId,
                method = method,
                body = body,
                url = url
            )
        }
    }
}
